utils/funcexp/func_mod.cpp

```diff
@@ -603,12 +600,11 @@ uint64_t Func_mod::getUIntVal(Row& row,
                 }
                 else
                 {
-                    int128_t scaleDivisor;
-                    datatypes::getScaleDivisor(scaleDivisor, d.scale);
-                    int128_t value = d.s128Value / scaleDivisor;
-                    int128_t lefto = d.s128Value % scaleDivisor;
-                    __float128 tmp = (__float128) (value % div) + (__float128) lefto / scaleDivisor;
-                    mod = datatypes::Decimal::getUInt64FromFloat128(tmp);
+                    auto intAndFract = d.getIntegralAndFractional<datatypes::TFloat128>();
+                    datatypes::TSInt128 integralRemainder = intAndFract.first % div;
+                    mod = static_cast<uint64_t>(integralRemainder.toTFloat128() +
+                                               intAndFract.second);
+
```

Comment:
`return static_cast<uint64_t>(doDecimal());`

Reply:
by the way, why aren't you fixing `Func_mod::getDecimalVal()`. It has exactly the same piece of code in here:
```
        // special case integer division
        else if (div.scale == 0)
        {
            int128_t scaleDivisor;
            datatypes::getScaleDivisor(scaleDivisor, d.scale);
            int128_t value = dividendInt / scaleDivisor;
            int128_t lefto = dividendInt % scaleDivisor;
            int128_t mod = (value % divInt) * scaleDivisor + lefto;
            retValue.s128Value = mod;
        }
```
it can be replaced to `return doDecimal();`

Reply:
Will do the latter.

Reply:
Implemented both suggestion.